### PR TITLE
Bootstrapping Generates Root Snapshot File

### PR DESCRIPTION
**Pull request by danuio**
Resolves https://github.com/dapperlabs/flow-go/issues/5286.

### CODE REVIEW

File: cmd/bootstrap/run/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package run
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+)
+
+// GenerateRootProtocolSnapshot generates the root protocol snapshot to be used
```
Comment (jordanschalm) [inline cmd/bootstrap/run/snapshot.go:8]: Do we need this?

**Review by jordanschalm — COMMENTED** [review]
For [this lint error](https://github.com/onflow/flow-go/pull/424/files#file-engine-common-rpc-convert-convert-go-L373), needs to do `serializable.Encodable()`. `inmem.Snapshot` implements `protocol.Snapshot` but is not directly serializable, `inmem.EncodableSnapshot` has all the exported fields needed to be serialized from another package.

For [this error](https://github.com/onflow/flow-go/pull/424/files#file-state-protocol-badger-validity-go-L124), it is fixed in this PR https://github.com/onflow/flow-go/pull/374

**Comment by jordanschalm** [comment]
We will also need to update the `testnet` code that uses this.

I have added some more detail to [the issue](https://github.com/dapperlabs/flow-go/issues/5286) regarding the final step of this ticket:
> Add an integration test identical to the MVP test, except one that starts from a post-genesis block

File: cmd/bootstrap/run/snapshot.go (hunk 1 above)
Comment (danuio) [reply cmd/bootstrap/run/snapshot.go:8]: Ahh no, we don't need this. I started this before noticing it was already implemented in the `inmem` package. I will remove it.

**Review by vishalchangrani — APPROVED** [review]

**Review by jordanschalm — APPROVED** [review]

**Pull request merged**